### PR TITLE
(#3462) Use Get-FileHash cmdlet where possible

### LINKED ISSUE #3462: Chocolatey is using checksum.exe which is slow and can't cope with larger than 2GB files
### Checklist

- [X] I confirm there are no unresolved issues reported on the [Chocolatey Status page](https://status.chocolatey.org).
- [X] I have verified this is the correct repository for opening this issue.
- [X] I have verified no other issues exist related to my problem.
- [X] I have verified this is not an issue for a specific package.
- [X] I have verified this issue is not security related.
- [X] I confirm I am using **official**, and **not** unofficial, or modified, Chocolatey products.

### What You Are Seeing?

Package fails to validate or install because download is larger than 2GB

### What is Expected?

Chocolatey should be able to cope with large files for calculating checksum/filehash

### How Did You Get This To Happen?

Tried to install package with downloaded file larger than 2GB eg- https://community.chocolatey.org/packages/resharper-platform/241.0.20240528.180551

### System Details

As per verifier environment

### Installed Packages

```bash
n/a
```

### Output Log

```bash
2024-06-10 01:03:39,185 4952 [INFO ] - Download of JetBrains.dotUltimate.2024.1.3.exe (2.55 GB) completed.
2024-06-10 01:03:42,544 4952 [DEBUG] - No runtime virus checking built into FOSS Chocolatey. Check out Pro/Business - https://chocolatey.org/compare
2024-06-10 01:03:42,637 4952 [DEBUG] - Verifying package provided checksum of '5342ede24c95a30261c0d43402c5cd3687d4b208233f028d7cb1ef7fe99a5a25' for 'C:\ProgramData\chocolatey\lib\resharper-platform\JetBrains.dotUltimate.2024.1.3.exe'.
2024-06-10 01:03:42,793 4952 [DEBUG] - Running Get-ChecksumValid -file 'C:\ProgramData\chocolatey\lib\resharper-platform\JetBrains.dotUltimate.2024.1.3.exe' -checksum '5342ede24c95a30261c0d43402c5cd3687d4b208233f028d7cb1ef7fe99a5a25' -checksumType 'sha256' -originalUrl 'https://download.jetbrains.com/resharper/dotUltimate.2024.1.3/JetBrains.dotUltimate.2024.1.3.exe' 
2024-06-10 01:03:43,091 4952 [DEBUG] - checksum.exe found at 'C:\ProgramData\chocolatey\helpers\..\tools\checksum.exe'
2024-06-10 01:03:43,123 4952 [DEBUG] - Executing command ['C:\ProgramData\chocolatey\helpers\..\tools\checksum.exe' -c="5342ede24c95a30261c0d43402c5cd3687d4b208233f028d7cb1ef7fe99a5a25" -t="sha256" -f="C:\ProgramData\chocolatey\lib\resharper-platform\JetBrains.dotUltimate.2024.1.3.exe"]
2024-06-10 01:40:08,840 4952 [WARN ] - Chocolatey timed out waiting for the command to finish. The timeout
 specified (or the default value) was '2700' seconds. Perhaps try a
 higher `--execution-timeout`? See `choco -h` for details.
2024-06-10 01:40:08,888 4952 [DEBUG] - Calling command ['"C:\Windows\System32\shutdown.exe" /a']
2024-06-10 01:40:09,955 4952 [DEBUG] - Command ['"C:\Windows\System32\shutdown.exe" /a'] exited with '1116'
2024-06-10 01:40:10,683 4952 [DEBUG] - Capturing package files in 'C:\ProgramData\chocolatey\lib\resharper-platform'
2024-06-10 01:40:10,745 4952 [WARN ] - Error computing hash for 'C:\ProgramData\chocolatey\lib\resharper-platform\JetBrains.dotUltimate.2024.1.3.exe'
 Hash will be special code for locked file or file too big instead.
 Captured error:
  The file is too long. This operation is currently limited to supporting files less than 2 gigabytes in size.
```
```


### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by flcdrg**
## Description Of Changes

Allow use of the Get-FileHash cmdlet when running with versions of PowerShell that include it

## Motivation and Context

Thsi should resolve the issue of checksums being unable to be verified for files greater than 2GB. It should also be faster. 

## Testing

Tested in Vagrant

## Change Types Made

* [x] Bug fix (non-breaking change).
* [ ] Feature / Enhancement (non-breaking change).
* [ ] Breaking change (fix or feature that could cause existing functionality to change).
* [ ] Documentation changes.
* [ ] PowerShell code changes.

## Change Checklist

* [ ] Requires a change to the documentation.
* [ ] Documentation has been updated.
* [ ] Tests to cover my changes, have been added.
* [ ] All new and existing tests passed?
* [x] PowerShell code changes: PowerShell v3 compatibility checked?

## Related Issue

Fixes chocolatey/checksum#23

### CODE REVIEW

**Comment by pauby** [comment]
Was `Get-FileHash` available in [PowerShell version 3.0](https://docs.chocolatey.org/en-us/chocolatey-components-dependencies-and-support-lifecycle#supported-powershell-versions)?

**Comment by flcdrg** [comment]
I'm assuming not, as it isn't listed in the 3.0 docs.

**Comment by pauby** [comment]
As we [still support PowerShell version 3.0](https://docs.chocolatey.org/en-us/chocolatey-components-dependencies-and-support-lifecycle#supported-powershell-versions), this isn't something we can pull in.

File: src/chocolatey.resources/helpers/functions/Get-CheckSumValid.ps1 [hunk 1]
```diff
@@ -189,27 +189,40 @@ Install-ChocolateyPackage
...
     }
 
-    $checksumExe = Join-Path "$helpersPath" '..\tools\checksum.exe'
-    if (!([System.IO.File]::Exists($checksumExe))) {
-        Update-SessionEnvironment
-        $checksumExe = Join-Path "$env:ChocolateyInstall" 'tools\checksum.exe'
-    }
-    Write-Debug "checksum.exe found at `'$checksumExe`'"
+    # Use Get-FileHash if available (added in Windows PowerShell 4.0)
+    if (Get-Command "Get-FileHash" -ErrorAction SilentlyContinue) {
```
Comment (TheCakeIsNaOH) [inline src/chocolatey.resources/helpers/functions/Get-CheckSumValid.ps1:193]: It probably would be significantly faster to check the PowerShell version, as `Get-Command` can be slow if the command does not exist (as it has to check the folders on the `$path` for native commands).

**Comment by pauby** [comment]
I've [moved the issue for this over to the checksum repository](https://github.com/chocolatey/checksum/issues/23). Rather than changing the way we calculate hashes, we're going to look at improving `checksum.exe` so won't be progressing with this PR.

We appreciate you submitting it, @flcdrg. I'll leave this open for any further comments before closing it.

**Pull request closed without merging**